Content.Server/Lathe/LatheSystem.cs

```diff
@@ -286,56 +241,62 @@ private void UpdateRunningAppearance(EntityUid uid, bool isRunning)
         /// </summary>
         private void UpdateInsertingAppearance(EntityUid uid, bool isInserting, Color? color = null)
         {
-            if (!TryComp<AppearanceComponent>(uid, out var appearance))
-                return;
-
-            appearance.SetData(LatheVisuals.IsInserting, isInserting);
+            _appearance.SetData(uid, LatheVisuals.IsInserting, isInserting);
             if (color != null)
-                appearance.SetData(LatheVisuals.InsertingColor, color);
+                _appearance.SetData(uid, LatheVisuals.InsertingColor, color);
+        }
+
+        private void OnPowerChanged(EntityUid uid, LatheComponent component, PowerChangedEvent args)
+        {
+            if (!args.Powered)
+            {
+                if (TryComp<LatheProducingComponent>(uid, out var prodcomp))
+                    RemComp(uid, prodcomp);
+            }
+            else if (component.Recipe != null)
+            {
+                EnsureComp<LatheProducingComponent>(uid);
+            }
+            UpdateRunningAppearance(uid, args.Powered && HasComp<LatheProducingComponent>(uid));
+        }
```

Comment:
RemComp returns a bool so shouldn't need TryComp

Also couldn't you just check if there's a queue and run TryStartProducing instead of in the update?